pkg/spdx/document.go

```diff
@@ -527,7 +527,19 @@ func (d *Document) ValidateFiles(filePaths []string) ([]ValidationResults, error
 	if len(filePaths) == 0 {
 		logrus.Warn("ValidateFiles called with 0 paths")
 	}
-	if len(d.Files) == 0 {
+
+	// Collect all available files from all packages for verification
+	allFiles := make(map[string]*File)
+	for k, v := range d.Files {
+		allFiles[k] = v
+	}
+	for _, pkg := range d.Packages {
+		for _, file := range pkg.Files() {
+			allFiles[file.ID] = file
+		}
+	}
```

Comment:
OK, so this has a problem. An SPDX document can describe one or more elements. This means that one or more files or packages can be "attached" to the root of the doc. Like this:

```
doc
 | - PackageA (Name: dir1)
 | - PackageB (Name: dir2)
 | - PackageC (Name: dir3)
 | - FileA (Name: file1.txt)
 | - FileB (Name: file2.txt)
```
 in order to ensure the integrity and completeness of the files in dir2 , we need to first find the directory in one of the top-level elements and inspect the files against the files in Package B. Otherwise, we could run into name clashes or other problems. For example, if we have two README.md files in Package A and Package B.

Something like this:

```suggestion
	allFiles := make(map[string]*File)
	var pkg *spdx.Package
	// Search for the package describing the directory
	for _, p := range d.Packages {
		if p.Name == opts.dir {
			pkg = p
			break
		}
	}
	if pkg == nil {
		return results, errors.New("directory not found in SBOM packages")
	}
	for _, file := range pkg.Files() {
		allFiles[file.ID] = file
	}
```

For the same reason, we should remove this:
```golang
for k, v := range d.Files {
		allFiles[k] = v
	}
```
Files described by the document directory (attached at the top) are guaranteed not to be in the directory we are looking for.

Reply:
Got it and changed it as suggested. I think we have to assume that the current working dir is the package, then it works as intended.

Reply:
Tricky: If no packages are specified, then we probably should still fallback to the root files (the tests are written like this), so I modified that path again.

Reply:
OK, yes it makes sense. Thanks for adding this @saschagrunert !

Reply:
Regarding assuming the directory as the cwd, yes this is the behavior I think would mak more sense. At list until we bake this flag @cpanato is working on into this subcommand: https://github.com/kubernetes-sigs/bom/pull/123